libs/wire-api/src/Wire/API/Routes/Public/Brig.hs

```diff
@@ -73,6 +77,27 @@ type DeleteSelfResponses =
      RespondWithDeletionCodeTimeout
    ]
 
+type CheckPasswordExistsResposes =
+  '[ Respond 200 "Password is set." (),
+     PasswordIsNotSet
+   ]
+
+newtype PasswordIsNotSet
+  = PasswordIsNotSet
+      (Respond 404 "Password is not set." ())
+  deriving (IsResponse '[JSON], IsSwaggerResponse)
+
+type instance ResponseType PasswordIsNotSet = ()
+
+-- TODO(sandy): Not sure about this instance -- does false correspond to the
+-- password being set?
+instance AsUnion CheckPasswordExistsResposes Bool where
+  toUnion False = Z (I ())
+  toUnion True = S (Z (I ()))
+  fromUnion (Z (I x)) = False
+  fromUnion (S (Z (I x))) = True
+  fromUnion (S (S ns)) = case ns of
+
```

Comment:
There should be no need for this. If you follow the convention is that "error" or "negative" responses go first, then there is already an instance of `AsUnion` for this case, and you should be able to use
```haskell
'[ Respond 404 ...,
  Respond 200 ...
```
as the list of responses.

Reply:
Sorry, I meant `RespondEmpty` instead of `Respond`.